packages/overlay/src/integrations/sentry/components/insights/Queries.tsx

```diff
@@ -108,19 +120,26 @@ const Queries = ({ showAll }: { showAll: boolean }) => {
       </Table.Header>
       <Table.Body>
         {queriesData.map(query => (
-          <tr key={query.description} className="hover:bg-primary-900">
+          <tr
+            key={query.description}
+            onClick={() => handleRowClick(query)}
+            onKeyDown={e => handleRowKeyDown(e, query)}
+            tabIndex={0}
+            role="link"
+            className="hover:bg-primary-900 cursor-pointer"
+          >
             <td className="text-primary-200 w-2/5 truncate whitespace-nowrap px-6 py-4 text-left text-sm font-medium">
-              <TimeBar value={query.totalTime} maxValue={maxTime} title={query.description}>
-                <Link className="truncate hover:underline" to={`/insights/queries/${btoa(query.description)}`}>
-                  {query.description}
-                </Link>
+              <TimeBar value={query.totalTime} maxValue={maxTime} title={query.description} className="text-lime-500">
+                {query.description}
               </TimeBar>
             </td>
             <td className="text-primary-200 w-[15%] whitespace-nowrap px-6 py-4 text-right text-sm font-medium">
-              {getFormattedDuration(query.avgDuration)}
+              <span className={getSpanDurationClassName(query.totalTime)}>{getFormattedDuration(query.totalTime)}</span>
```

Comment:
Thought for follow up: a `<SpanDuration duration={query.totalTime} />` component?